src/read.rs

```diff
@@ -228,6 +230,50 @@ impl<R: Read+io::Seek> ZipArchive<R>
         })
     }
 
+    /// Extract a Zip archive into a directory.
+    ///
+    /// Paths are sanitized so that they cannot escape the given directory.
+    ///
+    /// # Platform-specific behaviour
+    ///
+    /// On unix systems permissions from the zip file are preserved, if they exist.
+    pub fn extract(&mut self, directory: &Path) -> ZipResult<()> {
+        for i in 0..self.len() {
+            let mut file = self.by_index(i)?;
+            let filepath = file.sanitized_name();
+
+            // `sanitized_name` should return a relative path
+            // otherwise there's a risk of directory traversal attacks
+            assert!(filepath.is_relative());
```

Comment:
Should this be a `debug_assert`? Presumably if this ever gets tripped, it's a bug in zip-rs not in the user's crate. It feels wrong that we're forcing an extra runtime check to make sure this crate doesn't have bug.

Reply:
Looking at it now I'm not sure it's useful at all, removed.